src/Contrib/Jaeger/SpanConverter.php

```diff
@@ -0,0 +1,190 @@
+
+        // Zipkin tags must be strings, but opentelemetry
+        // accepts strings, booleans, numbers, and lists of each.
+        if (is_array($value)) {
+            return join(',', array_map([$this, 'sanitiseTagValue'], $value));
+        }
+
+        // Floats will lose precision if their string representation
+        // is >=14 or >=17 digits, depending on PHP settings.
+        // Can also throw E_RECOVERABLE_ERROR if $value is an object
+        // without a __toString() method.
+        // This is possible because OpenTelemetry\Trace\Span does not verify
+        // setAttribute() $value input.
+        return (string) $value;
+    }
+
+    private function buildTags(array $tagPairs): array
+    {
+        $tags = [];
+        foreach ($tagPairs as $key => $value) {
+            $tags[] = $this->buildTag($key, $value);
+        }
+
+        return $tags;
+    }
+
+    private function buildTag(string $key, string $value): Tag
+    {
+        return new Tag([
+            'key' => $key,
+            'vType' => TagType::STRING,
+            'vStr' => $value,
+        ]);
+
+        //TODO - determine if any of this additional handling is still needed, or if it's all dead-code (i.e. $value must be a string at this point always)
+
+        // if (is_bool($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::BOOL,
+        //         'vBool' => $value,
+        //     ]);
+        // } elseif (is_string($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::STRING,
+        //         'vStr' => $value,
+        //     ]);
+        // } elseif (null === $value) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::STRING,
+        //         'vStr' => '',
+        //     ]);
+        // } elseif (is_integer($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::LONG,
+        //         'vLong' => $value,
+        //     ]);
+        // } elseif (is_numeric($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::DOUBLE,
+        //         'vDouble' => $value,
+        //     ]);
+        // }
+
+        // error_log('Cannot build tag for ' . $key . ' of type ' . gettype($value));
+
+        // throw new \Exception('unsupported tag type');
+    }
+
+    private static function convertOtlpToJaegerIds(SpanDataInterface $span): array
+    {
+        if (PHP_INT_SIZE < 8) {
+            $humanReadableIntSize = PHP_INT_SIZE*8;
+
+            throw new RuntimeException("Integrating with Jaeger requires usage of 64 bit integers, but your current platform is $humanReadableIntSize bit. See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#ids for more information.");
```

Comment:
I think this belongs in a constructor (which you'll need to add) - better to fail fast if we can.

Reply:
Sounds good to me, moved it into one